ql/src/java/org/apache/hadoop/hive/ql/plan/ReplTxnWork.java

```diff
@@ -92,6 +120,18 @@ public ReplTxnWork(String dbName, String tableName, List<String> partNames,
     this.operation = type;
   }
 
+  public ReplTxnWork(String dbName, String tableName, List<String> partNames,
```

Comment:
have 2 constructors. one with the dumpDirectory and metricCollector and one without. That way you don't need to change existing code